### PR TITLE
BGE: custom lookup mappings

### PR DESCRIPTION
**Pull request by bethbrains**
--Custom lookup mappings are supported in BGE.

# Critical Changes

# Changes
- You can now include Lookup fields in your Batch Gift Entry batches. 

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by bethbrains** [comment]
**lurch: attach W-029639

File: src/classes/BGE_ConfigurationWizard_CTRL.cls [hunk 1]
```diff
@@ -191,38 +191,40 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
...
             'PHONE',
-            'URL'
+            'URL',
+            'REFERENCE'
         };
 
         // these fields have been whitelisted since they have special handling in BDI
         Set<String> allowedFields = new Set<String>{
-            'DonationCampaignImported__c'
+                'Donation_Record_Type_Name__c'
```
Comment (bethbrains) [inline src/classes/BGE_ConfigurationWizard_CTRL.cls]: This pattern is no longer needed for DonationCampaignImported__c, but is still needed for Donation Record Type, which came in a separate sibling branch. Added it here so it wasn't an empty set in this branch.

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007fPNuQAM" target="_blank">W-029639</a>

File: src/classes/BGE_DataImportBatchEntry_CTRL.cls [hunk 2]
```diff
@@ -347,40 +346,43 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
...
-                            String fieldLabel = dataImportDFR.name + 'label';
-                            String fieldName = dataImportDFR.name + 'link';
-                            linkCol.fieldName = fieldName;
-                            linkCol.readOnly = true;
-                            linkCol.required = field.requiredInEntryForm;
-                            linkCol.type = 'url';
-                            linkCol.typeAttributes = '{"label":{"fieldName":"'+fieldLabel+'"},"target":"_blank"}';
-                            columns.add(linkCol);
+                        
+                        if (dataImportDFR.type == Schema.DisplayType.Reference) {
```
Comment (ptewson-sfdo) [inline src/classes/BGE_DataImportBatchEntry_CTRL.cls]: 😁

File: src/classes/BGE_ConfigurationWizard_CTRL.cls [hunk 3]
```diff
@@ -191,38 +191,40 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
             'EMAIL',
             'MULTIPICKLIST',
             'PHONE',
-            'URL'
+            'URL',
+            'REFERENCE'
```
Comment (ptewson-sfdo) [inline src/classes/BGE_ConfigurationWizard_CTRL.cls]: This is a nice-to-have but would it be good to switch these out for the Schema.DisplayType enum values (or whatever the appropriate source is?)

File: src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntry.cmp [hunk 4]
```diff
@@ -44,6 +44,7 @@
     <aura:attribute name="data" type="List" default="[]" description="Data Import records that appear in the table" access="public"/>
     <aura:attribute name="errors" type="Map" description="Dry run errors to display on the lightning:dataTable" access="public"/>
     <aura:attribute name="columns" type="List" description="Columns for the table" access="public"/>
+    <aura:attribute name="referenceFields" type="List" description="Reference fields used in this batch" access="public"/>
```
Comment (ptewson-sfdo) [inline src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntry.cmp]: Does this need to be public? (Does 'columns' for that matter?)

Comment (bethbrains) [reply src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntry.cmp]: probably not!

File: src/classes/BGE_ConfigurationWizard_CTRL.cls (hunk 3 above)
Comment (bethbrains) [reply src/classes/BGE_ConfigurationWizard_CTRL.cls]: Yes, true. Good refactor.

**Pull request merged**